snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -569,7 +579,7 @@ def prime(self, force=False) -> None:  # noqa: C901
                 root_path=self.primedir,
                 snap_base_path=self._snap_base_path)
         elif is_classic:
-            dynamic_linker = self._project_options.get_core_dynamic_linker()
+            dynamic_linker = linker
```

Comment:
At this point, this kind of needs to be the pull path to the linker, but because of [this](https://github.com/snapcore/snapcraft/pull/1993/files#diff-660a35e421f136410b0d3e5168c7a62cR557) it's only the basename. We then use it to patch the interpreter :stuck_out_tongue: .

Reply:
I've fixed this in the final way actually; a way that is independent of the base using the FHS for each arch